src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -2051,6 +2053,12 @@ internal void WriteStateFile()
             }
             else if (!String.IsNullOrEmpty(_stateFile) && _cache.IsDirty)
             {
+                if (FailIfNotIncremental)
```

Comment:
This isn't the only output of RAR, right? Why is this the only one we care about? Because it's the only one saved to disk?

Reply:
This is not a complete incremental implementation for RAR.  I only add this here as changes to cache file is hard to detect.  There are enough cases where the RAR's output is the same but the cache is modified.  Because of the added risk, I started a separate PR that addressing some of the RAR issues.  #8173